src/validationinterface.h

```diff
@@ -35,6 +36,8 @@ class CValidationInterface {
     virtual void Inventory(const uint256 &hash) {};
     virtual void ResendWalletTransactions(int64_t nBestBlockTime) {};
     virtual void BlockChecked(const CBlock&, const CValidationState&) {};
+    virtual void GetScriptForMining(CScript &script) {};
```

Comment:
Agreed. Maybe it would be most efficient by adding a argument to `generate()` and `setgenerate()` RPC calls where one could provide a output script to mine to?

Adding a validation interface implementation for non-wallet-mining could be over the top.